.github/workflows/rust.yml

```diff
@@ -10,14 +10,59 @@ env:
   CARGO_TERM_COLOR: always
 
 jobs:
-  lint:
+  formatting:
 
     runs-on: ubuntu-latest
 
     steps:
     - uses: actions/checkout@v2
-    - name: Lint
+
+    - name: Formatting Check
       run: cargo fmt --all -- --check
+      
+  clippy:
+
+    runs-on: ubuntu-latest
+
+    container:
+      image: ghcr.io/psychedelic/dank-explorer-ci:latest
+      credentials:
+        username: ${{ github.actor }}
+        password: ${{ secrets.PRIVATE_ACCESS_TOKEN }}
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Rust toolchain
+        run: .github/scripts/rust-toolchain.sh
+
+      - name: Target clean cache
+        id: target-clean-cache
+        uses: actions/cache@v2
+        with:
+          path: .github/target_clean
+          key: cache-${{ runner.os }}-target-clean-cache-${{ hashFiles('.github/target_clean') }}
+
+      - name: Target dir cache
+        id: target-dir-cache
+        uses: actions/cache@v2
+        with:
+          path: target
+          key: cache-${{ runner.os }}-target-dir-cache-${{ hashFiles('cargo.lock') }}
+
+      - name: Target dir clean
+        # if: steps.target-dir-cache.outputs.cache-hit != 'true' || steps.target-clean-cache.outputs.cache-hit != 'true'
+        run: cargo clean
+
+      - name: Build
+        run: node build.js
```

Comment:
Why is this build needed here?

Reply:
`cargo clippy` builds & checks the program which requires the whole build flow. It would probably work fine if we made it part of the testing run as a stage, or somehow cache the build so the test runner doesn't need to rebuild.

Reply:
I think making this part of the test makes sense. 

One more thing I planned to cleanup is this:
- name: Build
        run: node build.js
- name: Cargo Test

this 2 tests use different compiler flags, so we are rebuilding from scratch twice